src/status_im2/common/universal_links.cljs

```diff
@@ -183,6 +176,55 @@
               {:db (dissoc db :universal-links/url)}
               (handle-url url))))
 
+(rf/reg-event-fx
+ :universal-links/generate-profile-url
+ (schema/instrument
+  ::generate-profile-url
+  (fn [{:keys [db] :as _cofx} [{:keys [public-key cb]}]]
+    (let [profile?   (not public-key)
+          ens-name?  (if profile?
+                       (get-in db [:profile/profile :ens-name?])
+                       (get-in db [:contacts/contacts public-key :ens-name]))
+          public-key (if profile? (get-in db [:profile/profile :public-key]) public-key)]
+      {:json-rpc/call
+       [{:method     (if ens-name? "wakuext_shareUserURLWithENS" "wakuext_shareUserURLWithData")
+         :params     [public-key]
+         :on-success #(do (rf/dispatch [:universal-links/save-profile-url public-key %])
```

Comment:
Quick one from the CSG https://guide.clojure.style/#no-multiple-forms-fn-literals